src/js/App/GlobalFilter/TagsModal.js

```diff
@@ -36,66 +53,113 @@ const TagsModal = ({ isOpen, filterTagsBy, onApplyTags, toggleModal, selectedTag
+                  namespace,
+                  key,
+                  value,
+                  selected: tagsSelected?.find?.(({ id } = {}) => id === `${namespace}/${key}=${value}`),
+                  cells: [key, value, namespace],
+                }))
+              ),
+            ]
+          : [[]]),
+        ...(sidLoaded
+          ? [
+              flatMap(sids, ({ tags }) =>
+                tags?.map(({ tag: { key, namespace } } = { tag: {} }) => ({
+                  namespace,
+                  id: key,
+                  key,
+                  selected: sidsSelected?.find?.(({ id } = {}) => id === key),
+                  cells: [key],
+                }))
+              ),
+            ]
+          : [[]]),
+      ]}
+      loaded={[tagsLoaded, sidLoaded]}
       width="50%"
       isOpen={isOpen}
       toggleModal={(_e, isSubmit) => {
-        setSelected([]);
+        setSidsSelected([]);
+        setTagsSelected([]);
         setFilterBy('');
         toggleModal(isSubmit);
       }}
       filters={[
-        {
-          label: 'Tags filter',
-          placeholder: 'Filter tags',
-          value: 'tags-filter',
-          filterValues: {
-            value: filterBy,
-            onChange: (_e, value) => {
-              setFilterBy(() => value);
-              debounceGeTags(value);
+        [
+          {
+            label: 'Tags filter',
+            placeholder: 'Filter tags',
+            value: 'tags-filter',
+            filterValues: {
+              value: filterBy,
+              onChange: (_e, value) => {
+                setFilterBy(() => value);
+                debounceGeTags(value);
+              },
             },
           },
-        },
+        ],
       ]}
-      onUpdateData={(pagination) =>
-        dispatch(
-          fetchAllTags(
-            {
-              registeredWith: filterScope,
-              activeTags: selectedTags,
-              search: filterBy,
-            },
-            pagination
-          )
-        )
-      }
-      columns={[{ title: 'Name' }, { title: 'Value' }, { title: 'Tag sources' }]}
-      onSelect={(selected) => setSelected(selected)}
-      selected={selected}
-      onApply={() => onApplyTags(selected)}
-      title="All tags"
+      onUpdateData={[
+        (pagination) =>
+          dispatch(
+            fetchAllTags(
+              {
+                registeredWith: filterScope,
+                activeTags: selectedTags,
+                search: filterBy,
+              },
+              pagination
+            )
+          ),
+        (pagination) =>
+          dispatch(
+            getAllSIDs(
+              {
+                registeredWith: filterScope,
+                activeTags: selectedTags,
+                search: filterBy,
+              },
+              pagination
+            )
+          ),
+      ]}
+      columns={[[{ title: 'Name' }, { title: 'Value' }, { title: 'Tag sources' }], [{ title: 'Value' }]]}
```

Comment:
There two columns with a `Value` title. Is that by design?

Reply:
That's actually two arrays of columns, one has 3 columns the other one has just `Value`